Large abelian groups and commutative monoids (#1602)
UniMath/SymmetryBookFormalization: src/group-theory/large-abelian-groups.lagda.md

open import group-theory.large-groups
open import group-theory.large-monoids
```

</details>

## Idea

A {{#concept "large abelian group" Agda=Large-Ab}} is a
[large group](group-theory.large-groups.md) whose binary operation is
commutative.

## Definition

```agda
record Large-Ab (α : Level → Level) (β : Level → Level → Level) : UUω where

  constructor
    make-Large-Ab

  field
    large-group-Large-Ab : Large-Group α β

  type-Large-Ab : (l : Level) → UU (α l)
  type-Large-Ab = type-Large-Group large-group-Large-Ab

  set-Large-Ab : (l : Level) → Set (α l)
  set-Large-Ab = set-Large-Group large-group-Large-Ab

  add-Large-Ab :
    {l1 l2 : Level} → type-Large-Ab l1 → type-Large-Ab l2 →
    type-Large-Ab (l1 ⊔ l2)
  add-Large-Ab = mul-Large-Group large-group-Large-Ab

  zero-Large-Ab : type-Large-Ab lzero
  zero-Large-Ab = unit-Large-Group large-group-Large-Ab

  neg-Large-Ab : {l : Level} → type-Large-Ab l → type-Large-Ab l
  neg-Large-Ab = inv-Large-Group large-group-Large-Ab

  field
    commutative-add-Large-Ab :
      {l1 l2 : Level} → (x : type-Large-Ab l1) → (y : type-Large-Ab l2) →
      add-Large-Ab x y ＝ add-Large-Ab y x

open Large-Ab public
```

## Properties

> ### The similarity relation of a large commutative monoid

Reviewer: ```suggestion
### The similarity relation of a large abelian group
```

    {l1 l2 l3 : Level} (x : type-Large-Ab G l1) (y : type-Large-Ab G l2) →
    add-Large-Ab G (raise-Large-Ab l3 x) y ＝
    raise-Large-Ab l3 (add-Large-Ab G x y)
  raise-left-add-Large-Ab = raise-left-mul-Large-Group (large-group-Large-Ab G)

  raise-right-add-Large-Ab :
    {l1 l2 l3 : Level} (x : type-Large-Ab G l1) (y : type-Large-Ab G l2) →
    add-Large-Ab G x (raise-Large-Ab l3 y) ＝
    raise-Large-Ab l3 (add-Large-Ab G x y)
  raise-right-add-Large-Ab =
    raise-right-mul-Large-Group (large-group-Large-Ab G)

  raise-left-unit-law-Large-Ab :
    {l1 l2 : Level} (x : type-Large-Ab G l1) →
    add-Large-Ab G (raise-zero-Large-Ab l2) x ＝ raise-Large-Ab l2 x
  raise-left-unit-law-Large-Ab =
    raise-left-unit-law-Large-Group (large-group-Large-Ab G)

  raise-right-unit-law-Large-Ab :
    {l1 l2 : Level} (x : type-Large-Ab G l1) →
    add-Large-Ab G x (raise-zero-Large-Ab l2) ＝
    raise-Large-Ab l2 x
  raise-right-unit-law-Large-Ab =
    raise-right-unit-law-Large-Group (large-group-Large-Ab G)

  raise-unit-lzero-Large-Ab :
    raise-zero-Large-Ab lzero ＝ zero-Large-Ab G
  raise-unit-lzero-Large-Ab =
    raise-unit-lzero-Large-Group (large-group-Large-Ab G)
```

> ### The inverse of the identity is the identity

Reviewer: ```suggestion
### The negative of the identity is the identity
```

      add-Large-Ab G (add-Large-Ab G y x) (neg-Large-Ab G x) ＝
      raise-Large-Ab G l1 y
    cancel-right-add-diff-Large-Ab =
      cancel-right-mul-div-Large-Group (large-group-Large-Ab G)

    sim-cancel-right-add-diff-Large-Ab :
      {l1 l2 : Level} (x : type-Large-Ab G l1) (y : type-Large-Ab G l2) →
      sim-Large-Ab G
        ( add-Large-Ab G (add-Large-Ab G y x) (neg-Large-Ab G x))
        ( y)
    sim-cancel-right-add-diff-Large-Ab =
      sim-cancel-right-mul-div-Large-Group (large-group-Large-Ab G)

    cancel-right-diff-add-Large-Ab :
      {l1 l2 : Level} (x : type-Large-Ab G l1) (y : type-Large-Ab G l2) →
      add-Large-Ab G (add-Large-Ab G y (neg-Large-Ab G x)) x ＝
      raise-Large-Ab G l1 y
    cancel-right-diff-add-Large-Ab =
      cancel-right-div-mul-Large-Group (large-group-Large-Ab G)

    sim-cancel-right-diff-add-Large-Ab :
      {l1 l2 : Level} (x : type-Large-Ab G l1) (y : type-Large-Ab G l2) →
      sim-Large-Ab G
        ( add-Large-Ab G (add-Large-Ab G y (neg-Large-Ab G x)) x)
        ( y)
    sim-cancel-right-diff-add-Large-Ab =
      sim-cancel-right-div-mul-Large-Group (large-group-Large-Ab G)
```

> ### Addition of an element of a large abelian group is an embedding

Reviewer: ```suggestion
### Addition by an element of a large abelian group is an embedding
```
